model/task_lifecycle.go

```diff
@@ -501,6 +495,14 @@ func UpdateBuildAndVersionStatusForTask(taskId string, updates *StatusChanges) e
 		return errors.WithStack(err)
 	}
 
+	initialBuildStatus := b.Status
+	oldTaskCacheInfo := []string{}
```

Comment:
Do you still want this logging?

Reply:
Yes, this commit unblocks completing Event Driven Notifications, it doesn't deal with the core issue, which I'm still curious about. This logging and the other part lets us find that.